Userland/Libraries/LibGfx/Matrix.h

```diff
@@ -75,6 +75,70 @@ class Matrix {
         return product;
     }
 
+    constexpr Matrix operator/(T divisor) const
+    {
+        Matrix division;
+        for (size_t i = 0; i < N; ++i) {
+            for (size_t j = 0; j < N; ++j) {
+                division.m_elements[i][j] = m_elements[i][j] / divisor;
+            }
+        }
+        return division;
+    }
+
+    constexpr Matrix adjugate() const
+    {
+        if constexpr (N == 1)
+            return Matrix(1);
+
+        Matrix adjugate;
+        for (size_t i = 0; i < N; ++i) {
+            for (size_t j = 0; j < N; ++j) {
+                int sign = (i + j) % 2 == 0 ? 1 : -1;
+                adjugate.m_elements[j][i] = sign * first_minor(i, j);
+            }
+        }
+        return adjugate;
+    }
+
+    constexpr T determinant() const
+    {
+        if constexpr (N == 1) {
+            return m_elements[0][0];
+        } else {
+            T result = {};
+            int sign = 1;
+            for (size_t j = 0; j < N; ++j) {
+                result += sign * m_elements[0][j] * first_minor(0, j);
+                sign *= -1;
+            }
+            return result;
+        }
+    }
+
+    constexpr T first_minor(size_t skip_row, size_t skip_column) const
+    {
+        VERIFY(N > 1);
```

Comment:
We should probably `static_assert(..)` this so It errors out T compile time.

Reply:
Right - changed it!